src/biogeophys/IrrigationMod.F90

```diff
@@ -762,27 +819,69 @@ subroutine ApplyIrrigation(this, bounds, waterfluxbulk_inst)
     ! works correctly.
 
     associate( &
-         qflx_irrig_patch => waterfluxbulk_inst%qflx_irrig_patch , & ! Output: [real(r8) (:)] patch irrigation flux (mm H2O/s)
-         qflx_irrig_col   => waterfluxbulk_inst%qflx_irrig_col     & ! Output: [real(r8) (:)] col irrigation flux (mm H2O/s)
+         qflx_sfc_irrig_patch      => waterfluxbulk_inst%qflx_sfc_irrig_patch      , & ! Output: [real(r8) (:)] patch irrigation flux (mm H2O/s)
+         qflx_sfc_irrig_col        => waterfluxbulk_inst%qflx_sfc_irrig_col        , & ! Output: [real(r8) (:)] col irrigation flux (mm H2O/s)
+         qflx_gw_uncon_irrig_patch => waterfluxbulk_inst%qflx_gw_uncon_irrig_patch , & ! Output: [real(r8) (:)] patch unconfined groundwater irrigation flux (mm H2O/s)
+         qflx_gw_uncon_irrig_col   => waterfluxbulk_inst%qflx_gw_uncon_irrig_col   , & ! Output: [real(r8) (:)] col unconfined groundwater irrigation flux (mm H2O/s)
+         qflx_gw_con_irrig_patch   => waterfluxbulk_inst%qflx_gw_con_irrig_patch   , & ! Output: [real(r8) (:)] patch confined groundwater irrigation flux (mm H2O/s)
+         qflx_gw_con_irrig_col     => waterfluxbulk_inst%qflx_gw_con_irrig_col     , & ! Output: [real(r8) (:)] col confined groundwater irrigation flux (mm H2O/s)
+         qflx_irrig_drip_patch     => waterfluxbulk_inst%qflx_irrig_drip_patch     , & ! Output: [real(r8) (:)] patch drip irrigation flux (mm H2O/s)
+         qflx_irrig_drip_col       => waterfluxbulk_inst%qflx_irrig_drip_col       , & ! Output: [real(r8) (:)] col drip irrigation flux (mm H2O/s)
+         qflx_irrig_sprinkler_patch=> waterfluxbulk_inst%qflx_irrig_sprinkler_patch, & ! Output: [real(r8) (:)] patch sprinkler irrigation flux (mm H2O/s)
+         qflx_irrig_sprinkler_col  => waterfluxbulk_inst%qflx_irrig_sprinkler_col    & ! Output: [real(r8) (:)] col sprinkler irrigation flux (mm H2O/s)
          )
 
     do p = bounds%begp, bounds%endp
```

Comment:
Now that this loop is doing more work, I'd like to see it changed to use a filter for performance reasons. It seems like soilp would be an appropriate filter here (though it's possible that there's a more appropriate one). Similarly, the p2c calls could be changed to use their filter version (in which case you should use a column filter that is analogous to the patch filter used in this loop - e.g., soilc together with soilp).

It's possible that, if you make this change, you might also need to initialize some / all of these fluxes to 0 in initCold.